### PR TITLE
Small changes to build output

### PR DESCRIPTION
**Pull request by cipherboy**
I'd like to get feedback on this. Removed notifications are currently commented out because they don't appear to be errors or warnings.

Based on feedback, I'll either remove the lines completely, or consider adding a VERBOSE option to the build that extends output, but which is default disabled. 

This'll help make builds more informative by reducing the quantity of extraneous information; limiting it to Errors and warnings. 

Thoughts?

### CODE REVIEW

**Comment by redhatrises** [comment]
Personally like the idea. I think adding a VERBOSE option would be way better than commenting out or removing the capability to get the output.

**Comment by mpreisler** [comment]
@cipherboy and I also discussed having a tool that will tell you about missing coverage, identifiers, etc... instead of it ever showing up in build output. We could probably re-purpose profile-stats.py for this.

**Comment by jan-cerny** [comment]
I would also prefer verbose option.

File: cmake/SSGCommon.cmake [hunk 1]
```diff
@@ -104,18 +104,32 @@ macro(ssg_build_xccdf_unlinked PRODUCT)
...
+            COMMAND "${OPENSCAP_OSCAP_EXECUTABLE}" xccdf resolve -o "${CMAKE_CURRENT_BINARY_DIR}/xccdf-unlinked-resolved.xml" "${CMAKE_CURRENT_BINARY_DIR}/xccdf-unlinked-resolved.xml"
+            COMMAND env "PYTHONPATH=$ENV{PYTHONPATH}" "${PYTHON_EXECUTABLE}" "${SSG_BUILD_SCRIPTS}/add_stig_references.py" --disa-stig "${STIG_REFERENCE_FILE}" --unlinked-xccdf "${CMAKE_CURRENT_BINARY_DIR}/xccdf-unlinked-resolved.xml"
+            DEPENDS generate-internal-${PRODUCT}-shorthand.xml
+            DEPENDS "${CMAKE_CURRENT_BINARY_DIR}/shorthand.xml"
+            DEPENDS "${CMAKE_CURRENT_SOURCE_DIR}/transforms/shorthand2xccdf.xslt"
+            DEPENDS "${CMAKE_CURRENT_SOURCE_DIR}/transforms/constants.xslt"
+            DEPENDS "${SSG_SHARED_TRANSFORMS}/shared_constants.xslt"
+            COMMENT "[${PRODUCT}-content] generating xccdf-unlinked-resolved.xml"
+        )
+    endif()
```
[the remark this answers is not in the document]
Comment (jan-cerny) [reply cmake/SSGCommon.cmake]: Would a macro be a way?

[the remark this answers is not in the document]
Comment (cipherboy) [reply cmake/SSGCommon.cmake]: @yuumasato Good catch that the current version (in [master](https://github.com/OpenSCAP/scap-security-guide/blob/master/cmake/SSGCommon.cmake#L107-L118)) doesn't depend on `STIG_REFERENCE_FILE`. I'll fix that as part of updating this PR (though it'll definitely need an update to add a `verbose` option, which I might not get to today).

**Comment by mpreisler** [comment]
I am not in favor of a verbose option, I'd much rather have a tool that tells me coverage details about all products and profiles. I do not want to see that in build output because most of the time it's not my fault that it's there. I only want to see legit warnings and errors and if there are any it should mean I **need** to fix something.

File: build-scripts/combine_remediations.py [hunk 2]
```diff
@@ -144,11 +144,11 @@ def main():
...
                         )
                 else:
-                    sys.stderr.write("Skipping '%s' remediation script. "
+                    sys.stderr.write("Warning: Skipping '%s' remediation script. "
                                      "The platform identifier in the "
                                      "script is missing!\n" % (filename))
-    sys.stderr.write("Merged %d %s remediations.\n"
-                     % (included_fixes_count, args.remediation_type))
+    #sys.stderr.write("Merged %d %s remediations.\n"
+    #                 % (included_fixes_count, args.remediation_type))
```
Comment (mpreisler) [inline build-scripts/combine_remediations.py]: I think we should keep this, it confirms that stuff is working and the output is not overwhelming.

File: ssg/build_ovals.py [hunk 3]
```diff
@@ -272,6 +272,6 @@ def checks(env_yaml, oval_version, oval_dirs):
                     body.append(xml_content)
                     included_checks_count += 1
                     already_loaded[filename] = oval_version
-    sys.stderr.write("Merged %d OVAL checks.\n" % (included_checks_count))
+    #sys.stderr.write("Merged %d OVAL checks.\n" % (included_checks_count))
```
Comment (mpreisler) [inline ssg/build_ovals.py]: same here, I think this confirms that things are working and it's just one line

File: build-scripts/sds_move_ocil_to_checks.py [hunk 4]
```diff
@@ -191,8 +191,8 @@ def main():
             # Also update the ID replacing 'ecomp' with 'comp'
             move_ocil_content_from_ds_extended_component_to_ds_component(datastreamtree, oldocilhref)
 
-    else:
-        print("No extended-components, nothing to do...")
+    #else:
+    #    print("No extended-components, nothing to do...")
```
Comment (mpreisler) [inline build-scripts/sds_move_ocil_to_checks.py]: +100

File: build-scripts/unselect_empty_xccdf_groups.py [hunk 5]
```diff
@@ -140,7 +140,7 @@ def main():
 
         affected_profiles.append(profile_element.get("id"))
 
-    print("Unselected empty groups in %s." % (", ".join(affected_profiles)))
+    # print("Unselected empty groups in %s." % (", ".join(affected_profiles)))
```
Comment (mpreisler) [inline build-scripts/unselect_empty_xccdf_groups.py]: probably OK not to print this.

**Comment by mpreisler** [comment]
I left some comments, IMO if they are addressed this can be merged. This still keeps the:
```
WARNING: OVAL check 'file_permissions_sshd_private_key' was not found, removing <check-content> element from the XCCDF rule.
```

warnings which I think we should get rid of but they need to be replaced by an easy way to list missing OVALs using `profile-stats.py`

File: cmake/SSGCommon.cmake [hunk 6]
```diff
@@ -99,11 +99,25 @@ macro(ssg_build_shorthand_xml PRODUCT)
...
+        COMMAND "${XSLTPROC_EXECUTABLE}" --stringparam ssg_version "${SSG_VERSION}" --output "${CMAKE_CURRENT_BINARY_DIR}/xccdf-unlinked-resolved.xml" "${CMAKE_CURRENT_SOURCE_DIR}/transforms/shorthand2xccdf.xslt" "${CMAKE_CURRENT_BINARY_DIR}/shorthand.xml"
+        COMMAND "${OPENSCAP_OSCAP_EXECUTABLE}" xccdf resolve -o "${CMAKE_CURRENT_BINARY_DIR}/xccdf-unlinked-resolved.xml" "${CMAKE_CURRENT_BINARY_DIR}/xccdf-unlinked-resolved.xml"
+        DEPENDS generate-internal-${PRODUCT}-shorthand.xml
+        DEPENDS "${CMAKE_CURRENT_BINARY_DIR}/shorthand.xml"
+        DEPENDS "${CMAKE_CURRENT_SOURCE_DIR}/transforms/shorthand2xccdf.xslt"
+        DEPENDS "${CMAKE_CURRENT_SOURCE_DIR}/transforms/constants.xslt"
+        DEPENDS "${SSG_SHARED_TRANSFORMS}/shared_constants.xslt"
+        COMMENT "[${PRODUCT}-content] generating xccdf-unlinked-resolved.xml"
+    )
+add_custom_target(
```
Comment (mpreisler) [inline cmake/SSGCommon.cmake]: indents

Comment (cipherboy) [reply cmake/SSGCommon.cmake]: Oops, that one snuck by me. Updated!

**Comment by cipherboy** [comment]
@mpreisler said:
> warnings which I think we should get rid of but they need to be replaced by an easy way to list missing OVALs using profile-stats.py

I have personally found those warnings to be helpful when merging rhel6->linux_os, so I'm preferential to keeping them.

**Comment by mpreisler** [comment]
ACK

**Pull request merged**